pkg/tfshim/sdk-v2/provider2.go

```diff
@@ -349,7 +349,7 @@ func (p *planResourceChangeImpl) Apply(
 
 	resp, err := p.server.ApplyResourceChange(ctx, t, ty, cfg, st, pl, priv, meta)
 	if err != nil {
-		return nil, err
+		return resp, err
```

Comment:
Curious how does this type-check? resp is the result of ApplyResourceChange but the type is shim.InstanceState?

Reply:
Hm, `ApplyResourceChange` returns a `v2InstanceState2` which fulfils `shim.InstanceState` and is the same type we create on the following line.

I wonder why we recreate it there. Thanks for spotting this.